### PR TITLE
Implement AI Lab extension installation test, add playwright framework and tests structure

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
* Adds playwright testing framework and folder structure
* Implements basic AI Lab extension installation test

### Screenshot / video of UI
[podman-desktop-ai-lab-extension-installation.webm](https://github.com/containers/podman-desktop-extension-ai-lab/assets/16451875/a5debe24-5855-43cc-b9ec-eb4bef580c92)

### What issues does this PR fix or reference?
containers/podman-desktop-internal/issues/174

### How to test this PR?
* clone podman-desktop repository
* build podman in development mode:
    * yarn install --frozen-lockfile
    * yarn test:e2e:build
* return back to ai-studio repository
* yarn install --frozen-lockfile
* yarn test:e2e

### Additional context
* Currently the AI Lab extension is rendered inside a `<webview>` element, making it very difficult if not impossible to interact with it through DOM
* Current implementation is as far as I'm able to go with the playwright framework (it's possible I'm unaware of a workaround)

### CODE REVIEW

**Review by odockal — COMMENTED** [review]
Looks pretty much alright, address the PR check complains.

**Comment by ScrewTSW** [comment]
> Looks pretty much alright, address the PR check complains.

fixed the linter issues
Code_WSQj1Emb5y

**Review by odockal — APPROVED** [review]

**Comment by odockal** [comment]
LGTM. But it would be nice to got someone to check yarn.lock since there is quite a lot of changes.

File: .eslintrc.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "ecmaVersion": 12,
     "sourceType": "module",
     "project": [
-      "packages/*/tsconfig.json"
+      "packages/*/tsconfig.json",
+      "tests/playwright/tsconfig.json"
```
Comment (benoitf) [inline .eslintrc.json:23]:
```suggestion
      "tests/*/tsconfig.json"
```

File: tests/playwright/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "ailab-tests-playwright",
```
Comment (benoitf) [inline tests/playwright/package.json:2]:
```suggestion
  "name": "ai-lab-tests-playwright",
```

File: tests/playwright/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  "version": "0.0.1",
```
Comment (benoitf) [inline tests/playwright/package.json:3]:
```suggestion
  "version": "0.4.0-next",
```

File: tests/playwright/src/ai-lab-extension-installation.spec.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+let navigationBar: NavigationBar;
+let dashboardPage: DashboardPage;
+let settingsBar: SettingsBar;
+let extensionsPage: SettingsExtensionsPage;
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('podman-desktop-ai-lab-extension-installation');
```
Comment (benoitf) [inline tests/playwright/src/ai-lab-extension-installation.spec.ts:43]:
```suggestion
  pdRunner.setVideoAndTraceName('podman-desktop-extension-ai-lab-installation');
```
(to mimic repository name)

**Review by benoitf — COMMENTED** [review]
it's missing lines in 
https://github.com/containers/podman-desktop-extension-ai-lab/blob/main/.github/workflows/release.yaml#L59-L61

to update version of package.json

**Comment by ScrewTSW** [comment]
> it's missing lines in https://github.com/containers/podman-desktop-extension-ai-lab/blob/main/.github/workflows/release.yaml#L59-L61
> 
> to update version of package.json

implemented suggestion, added version fix

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
> Currently the AI Lab extension is rendered inside a <webview> element, making it very difficult if not impossible to interact with it through DOM

did you do the focus ?
@ScrewTSW https://github.com/microsoft/playwright/issues/9729#issuecomment-1646860598

**Comment by benoitf** [comment]
@ScrewTSW next step is to add a workflow to launch the tests on each PR ?

**Comment by ScrewTSW** [comment]
> > Currently the AI Lab extension is rendered inside a  element, making it very difficult if not impossible to interact with it through DOM
> 
> did you do the focus ? @ScrewTSW [microsoft/playwright#9729 (comment)](https://github.com/microsoft/playwright/issues/9729#issuecomment-1646860598)

Interesting workaround, thank you for the suggestion

**Pull request merged**